src/riscv/uarch.c

```diff
@@ -0,0 +1,22 @@
+#include <stdint.h>
+
+#include <riscv/api.h>
+
+void cpuinfo_riscv_decode_vendor_uarch(
+	uint32_t vendor_id,
+	uint32_t arch_id,
+	uint32_t imp_id,
+	enum cpuinfo_vendor vendor[restrict static 1],
+	enum cpuinfo_uarch uarch[restrict static 1]) {
+	/* The vendor ID is sufficient to determine the cpuinfo_vendor. */
+	switch(vendor_id) {
+		case cpuinfo_riscv_chipset_sifive:
+			*vendor = cpuinfo_vendor_sifive;
+			break;
```

Comment:
Please add `default:` case that feels in vendor with some default info and print warning

Reply:
Done - added default and warning log.